accepted/2020/net5/net5.md

```diff
@@ -0,0 +1,854 @@
+
+* `net5.0-ios`
+* `net5.0-ios12.0`
+
+A TFM without an OS version will be interpreted as the lowest OS version that
+was supported by the corresponding `netX.Y` version.
+
+> *Note: One assumption here is that developers can always compile for the
+> latest OS API set and run on older OS versions so long they guard method calls
+> correctly. This is how P/Invokes, WinRT, iOS, and Android bindings work
+> today.*
+
+For example, say that when .NET 5 ships, the version of iOS that we include
+bindings for is iOS 13. That would mean that `net5.0-ios` and `net5.0-ios13.0`
+mean the exact same thing. Now lets say that Apple ships a new iOS 14 before
+.NET 6 is ready. We'd release a new version of the .NET 5 SDK that adds support
+for iOS 14. On machines with that SDK `net5.0-ios` still means `net5.0-ios13.0`.
+
+Please note that this mapping is specific for `net5.0`. When .NET 6 ships, we
+could decide that `net6.0-ios` means `net6.0-ios14.0`. But even when you use the
+.NET 6 SDK, when targeting .NET 5, `net5.0-ios` also still means
+`net5.0-ios13.0`. So these mappings are immutable.
+
+We have done this to simplify the experience for application models where the OS
+version is largely irrelevant, for example WinForms and WPF. Whether or not the
+template will put an OS version in the `<TargetFramework>` is up to each
+application model. Based on conversations, it seems we'll be landing on this:
+
+| TFM             | Project file includes OS version
+|-----------------|---------------------------------
+| net5.0-android  | Yes
+| net5.0-ios      | Yes
+| net5.0-macos    | Yes
+| net5.0-tvos     | Yes
+| net5.0-watchos  | Yes
+| net5.0-windows  | No
+
+Please note that by being able to put the OS version into the TFM one can also
+multi-target between OS versions:
+
+```XML
+<PropertyGroup>
+  <TargetFrameworks>net5.0-ios13.0;net5.0-ios14.0</TargetFrameworks>
+</PropertyGroup>
+```
+
+However, it's a bit misleading to think of the OS version number as the version
+of the operating system you're running on. Rather, it's the operating system's
+API you're compiling for.
+
+### Mapping to properties
+
+These are the relevant MSBuild properties:
+
+| Property                              | Meaning                         | Examples                         |
+|---------------------------------------|---------------------------------|----------------------------------|
+| `TargetFramework` (TFM)               | The friendly name               | `net4`, `netcoreapp3.0`          |
+| `TargetFrameworkIdentifier` (TFI)     | The long name                   | `.NETFramework` or `.NETCoreApp` |
+| `TargetFrameworkVersion` (TFV)        | The version number              | `2`, `3.0`, `3.1`                |
+| `TargetFrameworkProfile` (TFP)        | The profile                     | `Client` or `Profile124`         |
+| `TargetPlatformIdentifier` (TPI)      | The OS platform                 | `iOS`, `Android`, `Windows`      |
```

Comment:
Will this use a different case than the TFM?

Reply:
Yes!

<details>
<summary>More info</summary>

`TFM` is `TargetFrameworkMoniker` which is `.NETFramework,Version=4.0,Profile=Client` which is a very long name that is written to assemblies as `TargetFrameworkAttribute`.

What you're referring to is `TargetFramework` which is a short name or a friendly name and those will be in small case. This one is an `Identifier` which should be in case similar to `TFI` mentioned above.

</details>

Reply:
What @Nirmal4G said